irohad/main/iroha_conf_loader.hpp

```diff
@@ -63,16 +65,38 @@ struct IrohadConfig {
   boost::optional<logger::LoggerManagerTreePtr> logger_manager;
   boost::optional<shared_model::interface::types::PeerList> initial_peers;
   boost::optional<UtilityService> utility_service;
+
+  // This is a part of cryto providers feature:
+  // https://github.com/MBoldyrev/iroha/tree/feature/hsm-utimaco.
+  // This brings unnecessary complexity, but the aim is that this config section
+  // should require no modifications from users when the feature branch is
+  // merged.
+  struct Crypto {
+    struct Default {
+      static char const *kName;
+      iroha::multihash::Type type;
+      std::optional<std::string> private_key;
+    };
+
+    using ProviderId = std::string;
+    using ProviderList = std::unordered_map<ProviderId, Default>;
```

Comment:
```suggestion
    using ProviderId = size_t;
    using ProviderList = std::vector<Default>;
```
It looks like map really complicates the config, while a list or vector will do the same job as well, e.g.
```
export IROHA_CRYPTO_PROVIDERS_0_CRYPTO_TYPE=ed25519_sha3_256
export IROHA_CRYPTO_PROVIDERS_0_PRIVATE_KEY=cc5013e43918bd0e5c4d800416c88bed77892ff077929162bb03ead40a745e88
export IROHA_CRYPTO_PROVIDERS_0_TYPE=default
export IROHA_CRYPTO_SIGNER=0
```

Reply:
this will 
- make users count the index of needed provider in crypto providers list
- rise questions like "is the first index 0 or 1"?
- bring complexity to enumeration across providers from environment and config file
- require some changes in the existing code

as for me, it does not worth reducing the number of config vars by 1